src/container-mapping.ts

```diff
@@ -158,29 +188,37 @@ export class ContainerMapping implements IMicrosoftSecurityDevOps {
                 timeout: 2500,
                 headers: {
                     'Content-Type': 'application/json',
-                    'Authorization': 'Bearer ' + bearerToken
-                },
-                data: data
+                    'Authorization': 'Bearer ' + bearerToken,
+                    'Content-Length': data.length
+                }
             };
             core.debug(`${options['method'].toUpperCase()} ${url}`);
-    
+
             const req = https.request(url, options, (res) => {
                 let resData = '';
                 res.on('data', (chunk) => {
                     resData += chunk.toString();
                 });
-    
+
                 res.on('end', () => {
                     core.debug('API calls finished. Time taken: ' + (new Date().getMilliseconds() - apiTime) + "ms");
-                    core.debug('Response: ' + resData);
+                    core.debug(`Status code: ${res.statusCode} ${res.statusMessage}`);
+                    core.debug('Response headers: ' + JSON.stringify(res.headers));
+                    if (resData.length > 0) {
+                        core.debug('Response: ' + resData);
+                    }
+                    if (res.statusCode < 200 || res.statusCode >= 300) {
+                        reject(`Received Failed Status code when calling url: ${res.statusCode} ${resData}`);
+                    }
                     resolve();
                 });
             });
-    
+
             req.on('error', (error) => {
                 reject(new Error(`Error calling url: ${error}`));
             });
-            
+
+            req.write(data);

```

Comment:
To my question earlier, does `req.write(...)` a JSON string turn it into the object? :)

Reply:
No, we send the string directly to the server, and the server deserializes it accordingly (https://nodejs.org/api/http.html#requestwritechunk-encoding-callback)
Did I answer your question? I might have misunderstood.